project/Build.scala

```diff
@@ -57,13 +57,29 @@ object Scrooge extends Build {
       "sonatype-public" at "https://oss.sonatype.org/content/groups/public"
     ),
 
-    ScoverageSbtPlugin.ScoverageKeys.coverageHighlighting := true,
+    ScoverageKeys.coverageHighlighting := true,
 
     libraryDependencies ++= Seq(
-      "org.scalatest" %% "scalatest" % "2.2.4" % "test",
-      "org.scalacheck" %% "scalacheck" % "1.12.2" % "test",
+      scalatestLib(scalaVersion.value) % "test",
+      scalacheckLib(scalaVersion.value) % "test",
       "junit" % "junit" % "4.12" % "test"
     ),
+
+    libraryDependencies := {
```

Comment:
Unfortunately, this didn't work. That syntax only changes the resolution path for the plugin itself. For example using the following in `plugins.sbt`:
```
addSbtPlugin("org.scoverage" % "sbt-scoverage" % "1.3.5", sbtVersion = "0.13", scalaVersion = "2.11")
```
would result in the following error:
```
[info] Resolving org.scoverage#sbt-scoverage;1.3.5 ...
[warn] 	module not found: org.scoverage#sbt-scoverage;1.3.5
[warn] ==== typesafe-ivy-releases: tried
[warn]   https://repo.typesafe.com/typesafe/ivy-releases/org.scoverage/sbt-scoverage/scala_2.11/sbt_0.13/1.3.5/ivys/ivy.xml
[warn] ==== sbt-plugin-releases: tried
[warn]   https://repo.scala-sbt.org/scalasbt/sbt-plugin-releases/org.scoverage/sbt-scoverage/scala_2.11/sbt_0.13/1.3.5/ivys/ivy.xml
...
```
The only configuration that works would be:
```
addSbtPlugin("org.scoverage" % "sbt-scoverage" % "1.3.5", sbtVersion = "0.13", scalaVersion = "2.10")
```
which gives us the same result as `addSbtPlugin` without the additional parameters.

However, the problem of the non-existent-for-the-current-version-of-scala, automatically injected library dependencies that are included in the modules as a result of the plugin remains. Right now, mapping over the libraryDependencies as done both here and in `util` are the best way that I found to resolve this so far.